drivers/user_permission_report/app/controllers/user_permission_report/warehouse_reports/users_controller.rb

```diff
@@ -0,0 +1,24 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module UserPermissionReport::WarehouseReports
+  class UsersController < ApplicationController
+    before_action :set_user
+    helper_method :group_associations
+
+    def show
+      render layout: 'ajax_modal_rails/content'
+    end
+
+    private def set_user
+      @user = User.includes(:roles, access_groups: group_associations).find(params[:id])
+    end
+
+    def group_associations
+      UserPermissionReport::WarehouseReports::ReportsController.all_associations.keys
```

Comment:
Ah, this is why you made the other a class method.  You should be able to just throw a `new` between ReportsController and all_associations.  Let's try anyway.

Reply:
I wonder about a third possibility: What if we move that class method to the Project model?

Reply:
You know me, I love fat models.  Makes sense to me, but probably the User model since those are associations on user.

Reply:
✅ We moved group_associations to the user model.